k2/python/k2/dense_fsa_vec.py

```diff
@@ -116,18 +116,19 @@ def __init__(self,
                                  device='cpu',
                                  dtype=torch.int32)
         row_splits[1:] = torch.tensor(last_frame_indexes) + 1
-        # minus one to exclude the fake row [0, -inf, -inf, ...]
-        duration = row_splits[1:] - row_splits[:-1] - 1
 
         row_splits = row_splits.to(device)
         self.dense_fsa_vec = _k2.DenseFsaVec(scores, row_splits)
         self.scores = scores  # for back propagation
-        self._duration = duration.to(device)
 
     @property
     def duration(self) -> torch.Tensor:
         '''Return the duration of each seq.
         '''
+        if not hasattr(self, '_duration'):
```

Comment:
I would prefer to cache the CPU version of this, in the object.
CUDA->CPU transfers really slow down the pipeline because they force it to wait to sync the device.

Reply:
Fixed.